can/io/player.py

```diff
@@ -88,6 +92,23 @@ def __new__(  # type: ignore
                 f'No read support for this unknown log format "{suffix}"'
             ) from None
 
+    @staticmethod
+    def unzip(zipfile: "can.typechecking.StringPathLike"):
+        """
+        Return the suffix and io object of the decompressed file.
+        """
+        real_suffix = pathlib.Path(zipfile).suffixes[-2].lower()
+        file = gzip.open(zipfile, "rt")
+
+        # Re-open in binary mode if file not readable.
+        try:
+            file.read()
```

Comment:
> When size is omitted or negative, the entire contents of the file will be read and returned; it’s your problem if the file is twice as large as your machine’s memory.

This could be problematic, right?

Reply:
Do we need a `mode` argument or shall we infer it from `real_suffix`?

Reply:
> Do we need a `mode` argument or shall we infer it from `real_suffix`

We could but I guess I was trying to avoid hard coding it. 

One alternative could be to move the readers `mode` out of `__init__` and make it a class attribute instead. This way we could call something like  `getattr(LogReader.message_readers[suffix], "mode")`. Maybe I'm overthinking this though!

Reply:
We can still do this later 😄 I really like your proposal since its elegantly simple!

Reply:
I ended up inferring `mode` from `real_suffix` as suggested but I definitely think we should implement this later!